File: Python/Day 2/task1.py
file = open("C:\\Source\\amortsell\\AdventOfCode2021\\Day 2\\input.txt", "r")
lines = file.readlines()

class Direction(object):
  pass

count = 0
last = ''
xPos = 0
yPos = 0
directions = []
for line in lines:
  dir = Direction()
  parts = line.split(' ')
  dir.action = parts[0]
  dir.length = int(parts[1])
  directions.append(dir)

for direction in directions:
  if direction.action == "forward":
    xPos += direction.length
  elif direction.action == "down":
    yPos += direction.length
  else:
    yPos -= direction.length

result = xPos * yPos
print(result)

File: Typescript/Day 2/task1.ts
/// <reference path="../../node_modules/@types\\node/fs.d.ts" />

import * as fs  from 'fs';
import { Instruction } from './Models/Instruction';
import { Position } from './Models/Position';

let instructions: Array<Instruction> = []; 
const text = fs.readFileSync('Day 2/input.txt', 'utf8').toString();
const lines = [...text.split('\n')];
instructions.push(...lines.map(l => { return { action: l.split(' ')[0], length: parseInt(l.split(' ')[1])}; }));

let position: Position = instructions.reduce((prev, curr) => {
  switch (curr.action) {
    case 'forward':
      prev.x += curr.length;
      break;
    case 'up':
      prev.y -= curr.length;
      break;
    default:
      prev.y += curr.length;
      break;
  }
  return prev;
}, { x: 0, y: 0 } as Position);
console.log(position.x * position.y);